src/python/torchdistx/gossip_grad.py

```diff
@@ -339,17 +359,18 @@ def gossip_grad_hook(state: GossipGraDState, grad: torch.Tensor):
         >>>  from torchdistx.gossip_grad import(
         >>>     GossipGraDState,
         >>>     Topology,
+        >>>     get_num_modules,
         >>>     gossip_grad_hook
         >>>  )
         >>>
         >>>  net = torch.nn.Linear(4, 10)
         >>>  fsdp_net = FSDP(net)
-        >>>  state = GossipGraDState()
+        >>>  state = GossipGraDState(num_modules=get_num_modules(fsdp_net))
         >>>  fsdp_net.register_comm_hook(state, gossip_grad_hook)
 
     """
     # Virtual topology changes every `state.gossip_period` step
-    if state.iter % state.gossip_period == 0:
+    if (state.iter // state.num_modules) % state.gossip_period == 0:
```

Comment:
maybe add a comment for why we do this (i.e. same state iter is incremented by multiple FSDP modules)